build/tasks/debug.js

```diff
@@ -61,7 +61,7 @@ module.exports = function (grunt) {
         if (func) {
           // FunctionExpression.BlockStatement.body(Array)
           var funcBody = func.body.body;
-          esprima.parse('blocks.debug && blocks.debug.checkArgs(' + toValueString(data) + ', Array.prototype.slice.call(arguments), {})').body.forEach(function (chunk) {
+          esprima.parse('blocks.debug && blocks.debug.checkArgs(__DEBUG_METHOD, Array.prototype.slice.call(arguments), {}); var __DEBUG_METHOD = ' + toValueString(data) + ';').body.forEach(function (chunk) {
```

Comment:
Ahh got it! We are unshifting the chunks into the function, which means we are adding the parsed `body` reversed. I'll add a `reverse` before the foreach so it's more easy to read.

Regarding the .jshintrc:
jshint doesn't even check the generated code, the error is shown (if ``_DEBUG_METHOD`` is removed when checking the source (e.g. ``src/mvc/Application.js``).

We might could "fix" this when we add something macro-ish build step like
```javascript
//@throwDebug('mesage', 'type')
// which would be translated to something 
blocks.debug.throwMessage('message', __DEBUG_METHOD || null, 'type');
```
I'm not sure if I like that more.
We could implemted as a regex & replace function on the emitted code or with travelling through the ast and search & replace the matching nodes.

I'm not sure which of the methods is easier and less error prone.

What do you think ?